Support for GPS in flight GUI

Mart4672 opened the pull request:
Is this the correct way to do a pull request? So like do the changes that I made on my UI testing branch go into the main branch if you approve the changes?

Mart4672 committed:
Update flight_gui.py

Mart4672 committed:
Create flight_gui.asv

Mart4672 committed:
Merge branch 'master' into UI_Testing

Mart4672 committed:
Update flight_gui.py

Mart4672 committed:
Update flight_gui.py

krame505 commented on drivers/flight_gui.asv:
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
```
Duplicate file?

krame505 commented on drivers/flight_gui.py:
```diff
@@ -20,11 +20,22 @@ def vector_Plot(x, y, name=None, *args, **kwd_args):
 dts = ([manager.DataType('temperature', float, units='deg C', thresholds=(-20, 80))] +
-       vector_DataType('magnetometer', float, thresholds=(-100, 100)) + # TODO: units
-       vector_DataType('gyro', float, thresholds=(-100, 100)) + # TODO: units
+       [manager.DataType('bmp_alt', float, units='m', thresholds=(-100, 80000))] +
```
Style nit: these can all be one list instead of appending a bunch of single-element lists.  i.e.
```python
[manager.DataType('bmp_alt', float, units='m', thresholds=(-100, 80000)),
 manager.DataType('gps_alt', float, units='m', thresholds=(-100, 80000)),
 ...
 manager.DataType('sats', float, units='#', thresholds=(-10, 169))]
```

krame505 reviewed (commented):
Yeah, you are doing this right.

Mart4672 committed:
Update flight_gui.py

Mart4672 committed:
Delete flight_gui.asv

Mart4672 committed:
Update flight_gui.py

Mart4672 committed:
Update flight_gui.py

Fixed unit limits for euler angle as they weren't showing up properly in UI

Mart4672 committed:
added vb1 variable

added variable for battery1 voltage

Mart4672 committed:
UI additions for testing pyro channels

added many things from our current static test driver .py to make this one have similar button, timing, and value sending capabilities

Mart4672 committed:
Update flight_gui.py

Mart4672 committed:
units, minor fixes

Added many more data values to be recieved (for debugging purposes mostly)... will do more testing later in the week to ensure GUI is ready 4 flight

Mart4672 committed:
Update flight_gui.py

Mart4672 committed:
reset button, plots, new data types

added a reset button (doesn't seem to work- it seems I have to manually power/unpower the flight pc to get it to reset properly- I think this is the case with the static test stand stuff too...)
now plotting a live xy from the launch point
changed order of plots a bit

krame505 committed:
Merge branch 'master' into UI_Testing

krame505 committed:
Rescale plots size

Mart4672 committed:
bg

background additions

Mart4672 committed:
Merge branch 'UI_Testing' of https://github.com/LPRD/TelemetryController into UI_Testing

krame505 committed:
Merge branch 'master' into UI_Testing

krame505 committed:
Merge branch 'master' into UI_Testing

Mart4672 committed:
bg colors and positioning

made bg color more uniform, moved positions of plotting and side bar area... still needs more work, not sure how to precisely move the whole right side properly

krame505 committed:
Avoid setting the background colors globally

Mart4672 committed:
old changes that I forgot to update

Mart4672 committed:
Merge branch 'UI_Testing' of https://github.com/LPRD/TelemetryController into UI_Testing

Mart4672 committed:
Minor Fixes

Added:
-system state label changes color now based on state
-minor color and position tweaks so that everything is visible

Mart4672 committed:
compatability testing

Trying to get the start button to work (no luck so far)

Mart4672 committed:
Revert "compatability testing"

This reverts commit 49b1a44d112ce4507fce8c61664138ecbdaa60ba.

Mart4672 committed:
Clean Slate Start Point

The GUI mostly works EXCEPT for the Start button (and related countdown, plotting, etc). This represent a jumping off point for the current Flight/Cricket Code

Mart4672 committed:
Fixed Start Button

I would like to have the status.config background change colors depending on the state (for a visual aid), but my attempt (update_state listener) didn't work. How would you do this Lucas?

krame505 committed:
Updated dist files
